src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -507,6 +507,11 @@ static int uecc_rng_wrapper(uint8_t * dest, unsigned int size)
     return (ret == 0) ? size : 0;
 }
 
+static int uECC_is_rng_set(void)
+{
+	return (uecc_rng_wrapper == uECC_get_rng()) ? 1 : 0;
+}
+
```

Comment:
This is not thread-safe and does not help solve the reentrency problem. It is a quick fix for now, but a better way is to use an atomic boolean flag with proper atomic intrinsics and memory barriers to make sure it can only be overridden once

The actual correct way is to make the PLATFORM set this properly at init, rather than each method here trying to "come behind" and do it. Otherwise it is not possible to override the random number generator

Reply:
Please fix as a follow-up